query.go

```diff
@@ -420,26 +420,30 @@ func (q *query) queryPeer(ctx context.Context, ch chan<- *queryUpdate, p peer.ID
 	ctx, span := internal.StartSpan(ctx, "IpfsDHT.QueryPeer")
 	defer span.End()
 
-	dialCtx, queryCtx := ctx, ctx
-
 	// dial the peer
-	if err := q.dht.dialPeer(dialCtx, p); err != nil {
+	if err := q.dht.dialPeer(ctx, p); err != nil {
 		// remove the peer if there was a dial failure..but not because of a context cancellation
-		if dialCtx.Err() == nil {
+		if ctx.Err() == nil {
```

Comment:
This is racy we should maybe change to `errors.Is`, I didn't in case this was done due to improper error wrapping.

Reply:
I agree that it would be better to change to `error.Is` if possible.

Reply:
Dropped